### ISSUE TITLE
Current Fridge temperature rounding error

### ISSUE THREAD

**Issue by MichelRabozee** [QuickSander/homebridge-mieleathome#24]
Hello, since some time, I see in the homebridge accessory page the fact that the current temperature of the fridge is 4.1° instead of 4°. The Home application displays 4.0°, only because it truncate to the 1/10th of unit.

The State API returns:
```
  "temperature": [
    {
      "value_raw": 400,
      "value_localized": 4,
      "unit": "Celsius"
    }
```
I guess this comes during the division of value_raw by 100, but I do not understand why 400/100 gives 4.05 ??? Maybe because the division is done on integers with a float result ? Maybe it is needed to first cast value_raw to float ? or truncate the result to an integer ?

![current fridge temperature-1](https://user-images.githubusercontent.com/291583/111029560-9255e780-83fd-11eb-8eeb-8dc231e4932f.jpg)
![current fridge temperature-2](https://user-images.githubusercontent.com/291583/111029561-92ee7e00-83fd-11eb-9c78-89aee61c78bb.jpg)
![current fridge temperature-3](https://user-images.githubusercontent.com/291583/111029593-c8936700-83fd-11eb-8aa7-4587ae1610d5.jpg)

**Comment by QuickSander** [comment]
In javascript all numbers are 64 bit floating points. The current code uses a floating point literal to divide `raw_value/100.0`.

Doing some tests in node js:
```
> 400/100.0
4
> 400/100
4
> 3/4
0.75
> 
```

> or truncate the result to an integer

Javascript does not recognise integers, so it cannot truncate it either. I could try rounding, but since above test shows 400/100 is already exactly 4, rounding seems useless.

**Comment by QuickSander** [comment]
Could you enable debug logging. That way I can confirm the result received from the API is actually 400 and not 405 the moment the plugin retrieved it.

I don't think this is true, but I want to be 100% sure the value from the API doesn't slightly fluctuate since your fridge is controlling the temp within limits.

**Comment by MichelRabozee** [comment]
I did some tests, and really, I cannot understand how Homebridge is getting 4.05 instead of the 4.0 it received from the plugin.
This is a deep mystery. And it happens only in the Homebridge UI  and the Eve app.
Not really important I guess

**Comment by MichelRabozee** [comment]
(still with 2.7.2, which exhibits also the 4.05 displayed as 4.1)
```
Starting to advertise 'Corniche Homebridge-Three 05F5' using ciao backend!
[13/03/2021 16:12:37] Homebridge v1.3.3 (Corniche Homebridge-Three) is running on port 51809.
[13/03/2021 16:12:37] [Miele] Discovered devices:  [
  'mac-001D63FFFE0203B2',
  '000127334912',
  'mac-001D63FFFE0203B4',
  '000116035470',
  'mac-00124B0005AB4D8B'
]
[13/03/2021 16:12:37] [Miele] Discovered device: id: mac-001D63FFFE0203B2, name: Four, model: Unknown Miele model
[13/03/2021 16:12:37] [Miele] Ignoring unsupported device. Device: "Four" with raw type value: 12.
[13/03/2021 16:12:37] [Miele] Discovered device: id: 000127334912, name: Taque, model: Unknown Miele model
[13/03/2021 16:12:37] [Miele] Ignoring unsupported device. Device: "Taque" with raw type value: 14.
[13/03/2021 16:12:37] [Miele] Discovered device: id: mac-001D63FFFE0203B4, name: Lave-vaisselle, model: Unknown Miele model
[13/03/2021 16:12:37] [Miele] Re-creating existing accessory with data from cache: Lave-vaisselle
[13/03/2021 16:12:37] [Miele] Lave-vaisselle: Adding temperature sensor.
[13/03/2021 16:12:37] [Miele] Discovered device: id: 000116035470, name: Lave-linge, model: Unknown Miele model
[13/03/2021 16:12:37] [Miele] Re-creating existing accessory with data from cache: Lave-linge
[13/03/2021 16:12:37] [Miele] Discovered device: id: mac-00124B0005AB4D8B, name: Frigo, model: Unknown Miele model
[13/03/2021 16:12:37] [Miele] Re-creating existing accessory with data from cache: Frigo
[13/03/2021 16:12:38] [Miele] Cache retired. Status update enforced.
[13/03/2021 16:12:38] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/mac-001D63FFFE0203B4/state
[13/03/2021 16:12:38] [Miele] Cache retired. Status update enforced.
[13/03/2021 16:12:38] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/mac-00124B0005AB4D8B/state
[13/03/2021 16:12:38] [Miele] Cache retired. Status update enforced.
[13/03/2021 16:12:38] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/000116035470/state
[13/03/2021 16:12:38] [Miele] Parsed CurrentHeatingCoolingState from API response: 2
[13/03/2021 16:12:38] [Miele] Parsed TargetHeatingCoolingState from API response: 2
[13/03/2021 16:12:38] [Miele] Parsed zone 1 Current Temperature from API response: 400 [C/100]
[13/03/2021 16:12:38] [Miele] Parsed zone 1 Target Temperature from API response: 400 [C/100]
[13/03/2021 16:12:38] [Miele] Parsed Temperature Unit from API response: Celsius (HomeKit value: 0)
[13/03/2021 16:12:38] [Miele] Parsed Active from API response: 1
[13/03/2021 16:12:38] [Miele] Parsed InUse from API response: 0
[13/03/2021 16:12:38] [Miele] Parsed Remaining Duration from API response: 0 [s]
[13/03/2021 16:12:38] [Miele] Parsed Active from API response: 0
[13/03/2021 16:12:38] [Miele] Parsed InUse from API response: 0
[13/03/2021 16:12:38] [Miele] Parsed Remaining Duration from API response: 0 [s]
[13/03/2021 16:12:50] [Miele] Cache retired. Status update enforced.
[13/03/2021 16:12:50] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/mac-001D63FFFE0203B4/state
[13/03/2021 16:12:50] [Miele] Cache retired. Status update enforced.
[13/03/2021 16:12:50] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/mac-00124B0005AB4D8B/state
[13/03/2021 16:12:50] [Miele] Cache retired. Status update enforced.
[13/03/2021 16:12:50] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/000116035470/state
[13/03/2021 16:12:50] [Miele] Parsed CurrentHeatingCoolingState from API response: 2
[13/03/2021 16:12:50] [Miele] Parsed TargetHeatingCoolingState from API response: 2
[13/03/2021 16:12:50] [Miele] Parsed zone 1 Current Temperature from API response: 400 [C/100]
[13/03/2021 16:12:50] [Miele] Parsed zone 1 Target Temperature from API response: 400 [C/100]
[13/03/2021 16:12:50] [Miele] Parsed Temperature Unit from API response: Celsius (HomeKit value: 0)
[13/03/2021 16:12:50] [Miele] Parsed Active from API response: 1
[13/03/2021 16:12:50] [Miele] Parsed InUse from API response: 0
[13/03/2021 16:12:50] [Miele] Parsed Remaining Duration from API response: 0 [s]
[13/03/2021 16:12:50] [Miele] Parsed Active from API response: 0
[13/03/2021 16:12:50] [Miele] Parsed InUse from API response: 0
[13/03/2021 16:12:50] [Miele] Parsed Remaining Duration from API response: 0 [s]
```

**Comment by QuickSander** [comment]
Hmm strange indeed when I assume the display was consistently stating 4,05.
Especially since the target temperature is divided by 100 using the exact same code. Difference is that target temperature explicitly has a step size of 1 degree set. Maybe that pushes Homebridge in another logic.
Maybe something inside of Homebridge.

**Comment by MichelRabozee** [comment]
I have the same feeling about the homebridge logic. What about putting a 1 degree step for the current temperature as well ? (if not yet done of course).

**Comment by MichelRabozee** [comment]
When I set the fridge target temperature to 3°, the display of the current temperature indicated 3.1 as well once the temperature was reached (and I am sur the device State returned 300). I will try again, in debug mode and come back.

**Comment by MichelRabozee** [comment]
Here we are, always .05 added to the real value...
![current temperature 3 - 1](https://user-images.githubusercontent.com/291583/111036129-f2f51c80-841d-11eb-9396-cca4ef9224a3.jpg)
![current temperature 3 - 2](https://user-images.githubusercontent.com/291583/111036132-f5577680-841d-11eb-9997-0ae7287568a3.jpg)

:
```
[13/03/2021 17:00:15] [Miele] Discovered devices:  [
  'mac-001D63FFFE0203B2',
  '000127334912',
  'mac-001D63FFFE0203B4',
  '000116035470',
  'mac-00124B0005AB4D8B'
]
[13/03/2021 17:00:15] [Miele] Discovered device: id: mac-001D63FFFE0203B2, name: Four, model: Unknown Miele model
[13/03/2021 17:00:15] [Miele] Ignoring unsupported device. Device: "Four" with raw type value: 12.
[13/03/2021 17:00:15] [Miele] Discovered device: id: 000127334912, name: Taque, model: Unknown Miele model
[13/03/2021 17:00:15] [Miele] Ignoring unsupported device. Device: "Taque" with raw type value: 14.
[13/03/2021 17:00:15] [Miele] Discovered device: id: mac-001D63FFFE0203B4, name: Lave-vaisselle, model: Unknown Miele model
[13/03/2021 17:00:15] [Miele] Re-creating existing accessory with data from cache: Lave-vaisselle
[13/03/2021 17:00:15] [Miele] Lave-vaisselle: Adding temperature sensor.
[13/03/2021 17:00:15] [Miele] Discovered device: id: 000116035470, name: Lave-linge, model: Unknown Miele model
[13/03/2021 17:00:15] [Miele] Re-creating existing accessory with data from cache: Lave-linge
[13/03/2021 17:00:15] [Miele] Discovered device: id: mac-00124B0005AB4D8B, name: Frigo, model: Unknown Miele model
[13/03/2021 17:00:15] [Miele] Re-creating existing accessory with data from cache: Frigo
[13/03/2021 17:00:16] [Miele] Cache retired. Status update enforced.
[13/03/2021 17:00:16] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/mac-001D63FFFE0203B4/state
[13/03/2021 17:00:16] [Miele] Cache retired. Status update enforced.
[13/03/2021 17:00:16] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/mac-00124B0005AB4D8B/state
[13/03/2021 17:00:16] [Miele] Cache retired. Status update enforced.
[13/03/2021 17:00:16] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/000116035470/state
[13/03/2021 17:00:17] [Miele] Parsed Active from API response: 1
[13/03/2021 17:00:17] [Miele] Parsed InUse from API response: 0
[13/03/2021 17:00:17] [Miele] Parsed Remaining Duration from API response: 0 [s]
[13/03/2021 17:00:17] [Miele] Parsed CurrentHeatingCoolingState from API response: 2
[13/03/2021 17:00:17] [Miele] Parsed TargetHeatingCoolingState from API response: 2
[13/03/2021 17:00:17] [Miele] Parsed zone 1 Current Temperature from API response: 300 [C/100]
[13/03/2021 17:00:17] [Miele] Parsed zone 1 Target Temperature from API response: 300 [C/100]
[13/03/2021 17:00:17] [Miele] Parsed Temperature Unit from API response: Celsius (HomeKit value: 0)
[13/03/2021 17:00:17] [Miele] Parsed Active from API response: 0
[13/03/2021 17:00:17] [Miele] Parsed InUse from API response: 0
[13/03/2021 17:00:17] [Miele] Parsed Remaining Duration from API response: 0 [s]
[13/03/2021 17:01:15] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/mac-001D63FFFE0203B4/state
[13/03/2021 17:01:15] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/000116035470/state
[13/03/2021 17:01:15] [Miele] Update called. Requesting: https://api.mcs3.miele.com/v1/devices/mac-00124B0005AB4D8B/state
```

Really weird...

**Comment by QuickSander** [comment]
> What about putting a 1 degree step for the current temperature as well ? (if not yet done of course).

Not done yet, but I have mixed feelings about this. It could make it that you will only see 1 degree increments. Plus the fact that the default property has a 0.1 degree step set by Homebridge itself. There was an issue in the past with negative temperatures in HomeBridge.

**Comment by MichelRabozee** [comment]
Hi, as far as I know, Fridge temperature does only handle integer values. But anyway, this is not a big issue, if you feel more confortable leaving as it is (and quite surely, this is just a Homebridge issue which will be fixed one day).

**Comment by QuickSander** [comment]
I think you are right, but still strange their api messages deci-Celsius. That hints towards fractions of 1 degree steps. (I would like to keep the temp sensor common between devices)

**Comment by MichelRabozee** [comment]
Hello Sander, I do not know what caused the fix, but with new 3.2.0 version of your plugin, the fridge temperature is now correct in the Homebridge UI (and still correct in Home)
![fridge_temp_OK_in_UI](https://user-images.githubusercontent.com/291583/114139640-9814ef00-990f-11eb-95ce-e3c7afaecfdd.jpg)
Excellent !
